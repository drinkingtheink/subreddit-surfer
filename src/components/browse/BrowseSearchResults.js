import React, { Component } from "react"
import SearchResult from "./SearchResult"
import PropTypes from 'prop-types'

class BrowseSearchResults extends Component {
  render() {
    return (
      <div className="search-results-stage">
        <section className="search-results">
          {this.props.searchResultsGroup.list.map((result, index) => (
            <SearchResult
              redditBaseUrl={this.props.redditBaseUrl}
              key={index}
              result={result.data}
            />
          ))}
        </section>
      </div>
    );
  }
}

BrowseSearchResults.propTypes = {
  redditBaseUrl: PropTypes.string,
  searchResultsGroup: PropTypes.object
}

export default BrowseSearchResults;
