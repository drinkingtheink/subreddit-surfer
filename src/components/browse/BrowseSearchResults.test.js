import React from 'react'
import { shallow } from 'enzyme'
import BrowseSearchResults from './BrowseSearchResults'
import { exampleSearchResultsGroups } from './test/exampleSearchResultsGroups'

const redditBaseUrl = 'https://www.reddit.com'
const exampleSearchResultGroup = exampleSearchResultsGroups[0]

const wrapper = shallow(<BrowseSearchResults 
  redditBaseUrl={ redditBaseUrl }
  searchResultsGroup={ exampleSearchResultGroup }
 />)

describe(`BrowseSearchResults =====================`, () => {

  it(`renders with props`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});