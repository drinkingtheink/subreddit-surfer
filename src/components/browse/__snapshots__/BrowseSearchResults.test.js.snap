// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrowseSearchResults ===================== renders with props 1`] = `ShallowWrapper {}`;
